com.playeveryware.eos/Runtime/Core/Utility/FileSystemUtility.cs

```diff
@@ -478,15 +499,235 @@ public static string ReadAllText(string path)
+        /// entries being retrieved.
+        /// </returns>
+        public static IEnumerable<string> GetFileSystemEntries(string path, string pattern, bool recursive = true)
+        {
+            SearchOption option = recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+
+            return Directory.EnumerateFileSystemEntries(path, pattern, option);
+        }
+
+        #endregion
+
+
+        #region Path Functionality
+
+        /// <summary>
+        /// Wrapper function for calls to Path.Combine. The reason this is
+        /// implemented here is to restrict usage of the System.IO namespace to
+        /// be within this file.
+        /// </summary>
+        /// <param name="paths">A variable number of path elements.</param>
+        /// <returns>
+        /// The result of calling System.IO.Path.Combine with the provided path
+        /// components.
+        /// </returns>
+        public static string CombinePaths(params string[] paths)
+        {
+            return Path.Combine(paths);
+        }
+
+        /// <summary>
+        /// Wrapper function for calls to Path.GetFullPath. The reason this is
+        /// implemented here is to restrict usage of the System.IO namespace to
+        /// be within this file.
+        /// </summary>
+        /// <param name="path">The path (relative or full).</param>
+        /// <returns>The fully-qualified path.</returns>
+        public static string GetFullPath(string path)
+        {
+            return Path.GetFullPath(path);
+        }
+
+        /// <summary>
+        /// Wrapper function for calls to Path.GetFileName. The reason this is
+        /// implemented here is to restrict usage of the System.IO namespace to
+        /// be within this file.
+        /// </summary>
+        /// <param name="path">The path of the file to get the name of.</param>
+        /// <returns>The filename component of the given path.</returns>
+        public static string GetFileName(string path)
+        {
+            return Path.GetFileName(path);
+        }
+
+        #endregion
+
+        #region Write Functionality
+
+        /// <summary>
+        /// Synchronously writes the given file contents to the given filepath,
+        /// optionally creating the directory structure if it does not already
+        /// exist.
+        /// </summary>
+        /// <param name="filePath">The path to write to.</param>
+        /// <param name="content">The contents to write to the file.</param>
+        /// <param name="createDirectory">
+        /// Whether or not to create the directory structure represented by the
+        /// filepath.
+        /// </param>
+        public static void WriteFile(string filePath, string content, bool createDirectory = true)
+        {
+            // Use discard in-order to call async function without awaiting.
+            _ = WriteFileInternal(filePath, content, createDirectory);
```

Comment:
Does this actually run it synchronously? It looks like it just kicks off the function, but doesn't wait for it. Would calling this function promise that it'll be done by the time another function runs?

Reply:
Good inference. You are correct that it is not guaranteed to be finished. Per an earlier comment, I've updated the implementation to do it properly.